### PR TITLE
[Dependency Scanning] Break out Swift Overlay dependencies into separate output category

### PR DESCRIPTION
**Pull request by artemcm**
Instead of being a part of `directDependencies` on a module dependency info, make them a separate array of dependency IDs for Swift Source and Textual modules.

This will allow clients to still distinguish direct module dependencies imported from a given module, versus dependencies added because direct/transitive Clang module dependencies have Swift overlays.

**Note**: This change does *not* remove overlay dependencies from `directDependencies` yet, just adds them as a separate field on the module details info and adds corresponding `libSwiftScan` API to query them separately. A followup change will remove overlay and bridging header dependencies from `directDependencies` once the clients have had a chance to adopt to this change.

### CODE REVIEW

File: include/swift/Option/Options.td [hunk 1]
```diff
@@ -1286,6 +1286,10 @@ def disable_clang_target : Flag<["-"], "disable-clang-target">,
   Flags<[NewDriverOnlyOption]>,
   HelpText<"Disable a separately specified target triple for Clang instance to use">;
 
+def explain_module_dependency : Separate<["-"], "explain-module-dependency">,
```
Comment (cachemeifyoucan) [inline include/swift/Option/Options.td:1289]: Is this option tested?

**Review by cachemeifyoucan — COMMENTED** [review]
The change looks good to me. Seems lack testing for new option to explain why overlay dependency needs to be listed separately.

File: include/swift/Option/Options.td (hunk 1 above)
Comment (artemcm) [reply include/swift/Option/Options.td:1289]: > Is this option tested?

Yeah, it will be, here:
https://github.com/apple/swift-driver/pull/1363

Comment (artemcm) [reply include/swift/Option/Options.td:1289]: The option is `NewDriverOnly` so only the definition lives here, the driver repo will actually exercise and test it.

**Review by cachemeifyoucan — APPROVED** [review]

**Comment by artemcm** [comment]
@swift-ci test

**Review by nkcsgexi — APPROVED** [review]
Makes sense to me!

**Comment by artemcm** [comment]
@swift-ci test Linux platform

**Pull request merged**